Blog.API/Blog.Infrastructure/Abstract/Interfaces/IUnitOfWork.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace Blog.Infrastructure.Abstract.Interfaces;
+
+public interface IUnitOfWork
+{
+    IGenericRepository<T> GenericRepository<T>() where T : class;
+    Task SaveChangesAsync();
+}
```

Comment:
Because we're not going to create new class every time now interface need to reworked. Maybe something like this:

```suggestion
public interface IUnitOfWork<T> where T : class
{
    IGenericRepository<T> GenericRepository { get; }
    Task SaveChangesAsync();
}
```